lib/cockpit-components-listing.jsx

```diff
@@ -17,257 +17,267 @@
+        var countDisplay = null;
 
-            if (this.state.expanded) {
-                if (this.props.count > 1)
-                    count_display = <span className="pull-right">{ this.props.count + " occurrences"}</span>;
-                var links = this.props.tabRenderers.map(function(itm, idx) {
-                    return (
-                        <li key={idx} className={ (idx === self.state.activeTab) ? "active" : ""} >
-                            <a href="#" onClick={ self.handleTabClick.bind(self, idx) }>{itm.name}</a>
-                        </li>
-                    );
-                });
-                var tabs = [];
-                var tabIdx;
-                var Renderer;
-                var renderer_data;
-                var row;
-                for (tabIdx = 0; tabIdx < this.props.tabRenderers.length; tabIdx++) {
-                    Renderer = this.props.tabRenderers[tabIdx].renderer;
-                    renderer_data = this.props.tabRenderers[tabIdx].data;
-                    if (tabIdx !== this.state.activeTab && !(tabIdx in this.state.loadedTabs))
-                        continue;
-                    row = <Renderer key={ this.props.tabRenderers[tabIdx].name } hidden={ (tabIdx !== this.state.activeTab) } {...renderer_data} />;
-                    if (tabIdx === this.state.activeTab)
-                        tabs.push(<div className="listing-ct-body" key={tabIdx}>{row}</div>);
-                    else
-                        tabs.push(<div className="listing-ct-body" key={tabIdx} hidden>{row}</div>);
-                }
+        var headerEntries = this.props.columns.map(function(itm) {
+            if (typeof itm === 'string' || itm === null || itm instanceof String || React.isValidElement(itm))
+                return (<td>{itm}</td>);
+            else if ('header' in itm && itm.header)
+                return (<th>{itm.name}</th>);
+            else if ('tight' in itm && itm.tight)
+                return (<td className="listing-ct-tight listing-ct-actions">{itm.name || itm.element}</td>);
+            else
+                return (<td>{itm.name}</td>);
+        });
 
-                var listingDetail;
-                if ('listingDetail' in this.props) {
-                    listingDetail = (
-                        <span className="listing-ct-caption">
-                            {this.props.listingDetail}
-                        </span>
-                    );
-                }
+        var expandToggle = null;
+        if (this.props.tabRenderers.length > 0) {
+            expandToggle = <td className="listing-ct-toggle" onClick={ allowNavigate?this.handleExpandClick:undefined }>
+                               <i className="fa fa-fw"></i>
+                           </td>;
+        }
+        var listingItemClasses = ["listing-ct-item"];
+        if (!allowNavigate)
+            listingItemClasses.push("listing-ct-nonavigate");
```

Comment:
I think the class `listing-ct-nonavigate` is also only used by the listing pattern and could be moved out of cockpit.css